test/cpp/end2end/BUILD

```diff
@@ -219,6 +219,7 @@ grpc_cc_test(
 
 grpc_cc_test(
     name = "end2end_test",
+    timeout = "long",
```

Comment:
I think ideally this long timeout would only get applied for the poll-cv cases, because the other pollers seem to be ok with the shorter timeout. I could also imagine that we might find out some other tests (possibly also only under poll-cv) might need the "long" timeout.

What do you think about centralizing this timeout config, and adding some helper into `bazel/grpc_build_system.bzl`, e.g. `get_timeout_for_test_case` (which for now would apply the "long" timeout basically only if it is this test under poll-cv) which we could use around https://github.com/grpc/grpc/blob/master/bazel/grpc_build_system.bzl#L161.

Reply:
A query of recent dbg RBE runs (from 1/10) shows that only end2end_test with poll-cv poller has the timeout issue. Modifying the grpc_cc_test() function to accept poller-specific timeouts requires changes of many other targets, making it impractical. This should be a one-f case, as other targets with poll-cv did not demonstrate similar timeout issues.